test/button_test.js

```diff
@@ -34,84 +128,61 @@ describe("Add-on Functional Tests", function() {
-    // the start
-    const testText = "about:test";
-
-    // write dummy value just in case testText is already in clipboard
-    await clipboardy.write("foobar");
-    await utils.copyUrlBar(driver);
-    const clipboard = await clipboardy.read();
-    assert(clipboard === testText);
-  });
+  it("should have copy paste working", async() => copyPasteTest(driver));
 
-  it("animation should not trigger on disabled page", async() => {
-    // navigate to a disabled page
-    driver.setContext(Context.CONTENT);
-    await driver.get("about:blank");
-    driver.setContext(Context.CHROME);
+  it("animation should not trigger on disabled page",
+    async() => disabledPageAnimationTest(driver));
 
-    await utils.copyUrlBar(driver);
-    const { hasClass, hasColor } = await utils.testAnimation(driver);
-    assert(!hasClass && !hasColor);
-  });
+  it("popup should not trigger on disabled page", async() => disabledPagePopupTest(driver));
 
-  it("animation should trigger on regular page", async() => {
-    // navigate to a regular page
-    driver.setContext(Context.CONTENT);
-    await driver.get("http://mozilla.org");
-    driver.setContext(Context.CHROME);
+  it("animation should trigger on regular page", async() => regularPageAnimationTest(driver));
 
-    await utils.copyUrlBar(driver);
-    const { hasClass, hasColor } = await utils.testAnimation(driver);
-    assert(hasClass && hasColor);
-  });
+  it("popup should trigger on regular page", async() => regularPagePopupTest(driver));
 
-  it("popup should trigger on regular page", async() => {
-    // navigate to a regular page
-    driver.setContext(Context.CONTENT);
-    await driver.get("http://mozilla.org");
-    driver.setContext(Context.CHROME);
+  it("should not trigger treatments if the share button is in the overflow menu",
+    async() => overflowMenuTest(driver));
 
-    await utils.copyUrlBar(driver);
-    const panelOpened = await utils.testPanel(driver);
-    assert(panelOpened);
-  });
+  // ADDON UNINSTALLED
+  it("should no longer trigger animation once uninstalled",
+    async() => uninstallAddonAnimationTest(driver, addonId));
 
-  it("should not trigger treatments if the share button is in the overflow menu", async() => {
-    const window = driver.manage().window();
-    const currentSize = await window.getSize();
-    await window.setSize(640, 480);
-    await utils.copyUrlBar(driver);
-    assert(!(await utils.testPanel(driver)));
-    await window.setSize(currentSize.width, currentSize.height);
-  });
+  it("should no longer trigger popup once uninstalled",
+    async() => uninstallAddonPopupTest(driver));
+});
 
-  // ADDON UNINSTALLED
-  it("should no longer trigger animation once uninstalled", async() => {
-    await utils.uninstallAddon(driver, addonId);
-    await utils.copyUrlBar(driver);
-    const { hasClass, hasColor } = await utils.testAnimation(driver);
-    assert(!hasClass && !hasColor);
+describe("Multiple Window Add-on Functional Tests", function() {
```

Comment:
I think "multiple" here isn't strict correct, since you still only have one window.